torch/_dynamo/variables/optimizer.py

```diff
@@ -50,6 +60,11 @@ def call_method(
                 self.value._init_group(*py_args, **py_kwargs)
                 self.map_sources_and_install_guards(tx)
                 self.update_list_args(tx, args, kwargs, py_args, py_kwargs)
+                # stash a weak_ptr to optimizer to invalidate code
+                # if the optimizer object dies
+                tx.store_global_weakref(self.get_global_name(), self.value)
```

Comment:
Could this be a more generic mechanism ?

Reply:
This is already used by a bunch of different components (dynamo for dict keys, optim parameters, torch function subclasses)